### PR TITLE
[build/release] support to incorporate musl compiled binaries into release

### PR DESCRIPTION
**Pull request by ZhiHanZ**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary
Support the following targets under release
1. x86_64-unknown-linux-musl
2. aarch64-unknown-linux-musl

By incorporating musl under regular release, we should guide users under environment such as RHEL7, RHEL8, centOS8
centOS7, arch etc to use static linked binary if they have incompatible release binary versions
## Changelog

- Build/Testing/CI
- Need Documentation (Need to add documentation to https://databend.rs)

## Related Issues
Fixes #https://github.com/datafuselabs/databend/issues/4513
Fixes #https://github.com/datafuselabs/databend/issues/3570

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: docker/build-tool/musl/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM rustembedded/cross:x86_64-unknown-linux-musl
+COPY ./openssl.sh /
```
Comment (Xuanwo) [inline docker/build-tool/musl/Dockerfile:2]: I really don't like the idea to compile openssl.

How about using `alpine` directly? https://github.com/datafuselabs/databend/pull/4536

Comment (ZhiHanZ) [reply docker/build-tool/musl/Dockerfile:2]: re-invent the alpine based build-kit is not easy, we could consider separate them into smaller tasks and iteratively replace current cross utilizations

**Review by Xuanwo — APPROVED** [review]
It's OK to make databend have a musl static releases first and than improve it.

**Review by PsiACE — APPROVED** [review]

File: docker/build-tool/openssl.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    # clean up
+    apt-get purge --auto-remove -y ${purge_list[@]}
+
+    popd
+
+    rm -rf $td
+    rm $0
+}
+
+main "${@}"
```
Comment (everpcpc) [inline docker/build-tool/openssl.sh:49]: newline at end of file

**Review by everpcpc — APPROVED** [review]

**Pull request merged**